source/pele/base_potential.h

```diff
@@ -15,7 +15,32 @@ namespace pele {
 class BasePotential {
 public:
     virtual ~BasePotential() {}
-
+    
+    /**
+     * Return the number of dimensions (box dimensions).
+     * Ideally this should be overloaded.
+     */
+    virtual size_t get_ndim()
+    {
+        throw std::runtime_error("BasePotential::get_ndim must be overloaded");
+    }
+    
+    /**
+     * Return the distance as measured by the distance policy. 
+     */
+    virtual void get_rij(double * const r_ij, double const * const r1, double const * const r2) const
+    {
+        throw std::runtime_error("BasePotential::get_rij must be overloaded");
+    }
+    
+    /**
+     * Return energy_gradient of iteraction. 
+     */
+    virtual double get_interaction_energy_gradient(double r2, double *gij, size_t atom_i, size_t atom_j) const
```

Comment:
ok I guess SimplePairwisePotential is a alredy the particles potential we
needed.

On Thu, May 14, 2015 at 11:21 AM, Jacob Stevenson <notifications@github.com>
wrote:

> In source/pele/base_potential.h
> <https://github.com/pele-python/pele/pull/130#discussion_r30311049>:
>
> > +    {
> > +        throw std::runtime_error("BasePotential::get_ndim must be overloaded");
> > +    }
> > +
> > +    /**
> > +     * Return the distance as measured by the distance policy.
> > +     */
> > +    virtual void get_rij(double * const r_ij, double const * const r1, double const * const r2) const
> > +    {
> > +        throw std::runtime_error("BasePotential::get_rij must be overloaded");
> > +    }
> > +
> > +    /**
> > +     * Return energy_gradient of iteraction.
> > +     */
> > +    virtual double get_interaction_energy_gradient(double r2, double *gij, size_t atom_i, size_t atom_j) const
>
> And you could require that a distance function is passed to compute rij.
> That's what we've done before
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pele-python/pele/pull/130/files#r30311049>.
>